### PR TITLE
Fix LGTM alerts

### PR DESCRIPTION
**Pull request by martinpitt**
See https://lgtm.com/projects/g/cockpit-project/bots/alerts/?mode=list

Unfortunately I didn't find a way to tell lgtm to actually *fail* their test status on new alerts... :cry:

### CODE REVIEW

**Comment by martinpitt** [comment]
Okay, this [fixes 8 alerts](https://lgtm.com/projects/g/cockpit-project/bots/rev/pr-a2646c734a06ee7140daaf88ec1afa10b46e309e), but there are [11 current ones](https://lgtm.com/projects/g/cockpit-project/bots/alerts/?mode=list). It seems it doesn't count the three alerts that I ignored? I might have messed up the [alert suppression](https://lgtm.com/help/lgtm/alert-suppression), possibly the missing "py/" prefix. Trying with that.

**Comment by martinpitt** [comment]
Okay, :man_shrugging: -- maybe it just doesn't consider ignored alerts as "fixed" then? Let's at least land these 11? I trigger a few tests, just to make sure the changes don't break anything, and also to validate https://github.com/cockpit-project/cockpituous/pull/517.

File: machine/machine_core/machine_virtual.py [hunk 1]
```diff
@@ -169,23 +169,21 @@ def _lock(self, start, step=1, force=False):
...
+                    fcntl.flock(lockf, fcntl.LOCK_NB | fcntl.LOCK_EX)
+                    sock.bind(("127.0.0.1", port))
+                    self.locked.append(lockf)
+                except IOError:
+                    if force:
+                        return port
+                    os.close(lockf)
+                    continue
+                else:
                     return port
```
Comment (allisonkarlitskaya) [inline machine/machine_core/machine_virtual.py:186]: Unrelated, but another cleanup to this too-complicated code:

Can you drop the `else:` clause by pulling the `return port` into the `try:` or just unindenting it?  The `except:` clause never returns.

You could also save a line by inverting the condition in the `except:` clause and allowing the fall-through to the `return`.

**Review by allisonkarlitskaya — APPROVED** [review]
Feel free to commit as is, or bonus points for the suggested change.

In either case, thanks for the cleanups :)

File: machine/machine_core/machine_virtual.py (hunk 1 above)
Comment (martinpitt) [reply machine/machine_core/machine_virtual.py:186]: Good point! I'm not done with bots anyway, I have some more cleanups in the pipe once the current set of entangled PRs lands. I'll do that as a follow-up!

**Pull request merged**